core/modules/layout/layout.admin.inc

```diff
@@ -1110,21 +1110,69 @@ function layout_settings_form_save_submit($form, &$form_state) {
     unset($_SESSION['layout_new_name']);
   }
 
-  // If this is a new layout, populate blocks from default layout.
+  // If this is a new layout, populate it with blocks from the default layout.
   if (!empty($layout->is_new)) {
     $default_layout_name = path_is_admin($layout->getPath()) ? 'admin_default' : 'default';
     $default_layout = layout_load($default_layout_name);
+    // Skip any regions in the default layout that do not contain any blocks.
+    $default_layout_regions = array_filter($default_layout->positions);
+
     // Get the template being used by this layout.
     $template_info = layout_get_layout_template_info($layout->layout_template);
-    foreach ($default_layout->positions as $region => $region_block_list) {
-      if (array_key_exists($region, $template_info['regions'])) {
-        foreach ($region_block_list as $uuid) {
-          if (isset($default_layout->content[$uuid])) {
-            $block = $default_layout->content[$uuid];
+    $template_is_flexible = !empty($template_info['flexible']);
+
+    if ($template_is_flexible) {
```

Comment:
We shouldn't need any special handling for flexible templates, let's take this change out.

Reply:
Agreed 👍🏼 ...and thank you 🙏🏼 ...I'll close this PR in favor of the other.